### PR TITLE
feat: Add new options for `upload-assets` plugin for compact and grouped view

### PR DESCRIPTION
**Pull request by neretin-trike**
Реализовано расширение плагина `upload-assets`, в которое были добавлены следующие опции:

`headerMessage` - Кастомное сообщение для заголовка в сообщении Pull Request'a.
`filter` - Фильтр для списка ассетов. Выполняет проверку с помощью метода `test` объекта RegEx.
`includeBotPrs` - Добавлять ли сообщение в Pull Requests. По умолчанию `true`.
`group`- Группировать список асссетов. Выполняет метод `exec` объекта RegEx и берет второй результат.
`compact` - Компактный вид сообщений Pull Request'ов . По умолчанию `false`.

<details>
  <summary>:baby_chick: Download canary assets:</summary>
  <blockquote>
    <details>
      <summary>color</summary>
      <blockquote>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_ios-swift--canary.605.5507511157.swift'>color_asdk_ios-swift--canary.605.5507511157.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_kotlin--canary.605.5507511157.kt'>color_asdk_kotlin--canary.605.5507511157.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_react-native--canary.605.5507511157.ts'>color_asdk_react-native--canary.605.5507511157.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_xml--canary.605.5507511157.xml'>color_asdk_xml--canary.605.5507511157.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_ios-swift--canary.605.5507511157.swift'>color_plasma_b2c_ios-swift--canary.605.5507511157.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_kotlin--canary.605.5507511157.kt'>color_plasma_b2c_kotlin--canary.605.5507511157.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_react-native--canary.605.5507511157.ts'>color_plasma_b2c_react-native--canary.605.5507511157.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_xml--canary.605.5507511157.xml'>color_plasma_b2c_xml--canary.605.5507511157.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.605.5507511157.swift'>color_sberHealth_ios-swift--canary.605.5507511157.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.605.5507511157.kt'>color_sberHealth_kotlin--canary.605.5507511157.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.605.5507511157.ts'>color_sberHealth_react-native--canary.605.5507511157.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_xml--canary.605.5507511157.xml'>color_sberHealth_xml--canary.605.5507511157.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_ios-swift--canary.605.5507511157.swift'>color_sbermarket_business_ios-swift--canary.605.5507511157.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_kotlin--canary.605.5507511157.kt'>color_sbermarket_business_kotlin--canary.605.5507511157.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_react-native--canary.605.5507511157.ts'>color_sbermarket_business_react-native--canary.605.5507511157.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_xml--canary.605.5507511157.xml'>color_sbermarket_business_xml--canary.605.5507511157.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.605.5507511157.swift'>color_sbermarket_ios-swift--canary.605.5507511157.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.605.5507511157.kt'>color_sbermarket_kotlin--canary.605.5507511157.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_ios-swift--canary.605.5507511157.swift'>color_sbermarket_metro_ios-swift--canary.605.5507511157.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_kotlin--canary.605.5507511157.kt'>color_sbermarket_metro_kotlin--canary.605.5507511157.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_react-native--canary.605.5507511157.ts'>color_sbermarket_metro_react-native--canary.605.5507511157.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_xml--canary.605.5507511157.xml'>color_sbermarket_metro_xml--canary.605.5507511157.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.605.5507511157.ts'>color_sbermarket_react-native--canary.605.5507511157.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_ios-swift--canary.605.5507511157.swift'>color_sbermarket_selgros_ios-swift--canary.605.5507511157.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_kotlin--canary.605.5507511157.kt'>color_sbermarket_selgros_kotlin--canary.605.5507511157.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_react-native--canary.605.5507511157.ts'>color_sbermarket_selgros_react-native--canary.605.5507511157.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_xml--canary.605.5507511157.xml'>color_sbermarket_selgros_xml--canary.605.5507511157.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_ios-swift--canary.605.5507511157.swift'>color_sbermarket_wlbusiness_ios-swift--canary.605.5507511157.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_kotlin--canary.605.5507511157.kt'>color_sbermarket_wlbusiness_kotlin--canary.605.5507511157.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_react-native--canary.605.5507511157.ts'>color_sbermarket_wlbusiness_react-native--canary.605.5507511157.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_xml--canary.605.5507511157.xml'>color_sbermarket_wlbusiness_xml--canary.605.5507511157.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_xml--canary.605.5507511157.xml'>color_sbermarket_xml--canary.605.5507511157.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_ios-swift--canary.605.5507511157.swift'>color_sberonline_ios-swift--canary.605.5507511157.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_kotlin--canary.605.5507511157.kt'>color_sberonline_kotlin--canary.605.5507511157.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_react-native--canary.605.5507511157.ts'>color_sberonline_react-native--canary.605.5507511157.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_xml--canary.605.5507511157.xml'>color_sberonline_xml--canary.605.5507511157.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.605.5507511157.swift'>color_sberprime_ios-swift--canary.605.5507511157.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.605.5507511157.kt'>color_sberprime_kotlin--canary.605.5507511157.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.605.5507511157.ts'>color_sberprime_react-native--canary.605.5507511157.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_xml--canary.605.5507511157.xml'>color_sberprime_xml--canary.605.5507511157.xml</a><br>
      </blockquote>
    </details>
    <details>
      <summary>shadow</summary>
      <blockquote>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.605.5507511157.ts'>shadow_sbermarket_react-native--canary.605.5507511157.ts</a><br>
      </blockquote>
    </details>
    <details>
      <summary>typo</summary>
      <blockquote>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.605.5507511157.swift'>typo_mage_ios-swift--canary.605.5507511157.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.605.5507511157.kt'>typo_mage_kotlin--canary.605.5507511157.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.605.5507511157.ts'>typo_mage_react-native--canary.605.5507511157.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.605.5507511157.swift'>typo_plasma_ios-swift--canary.605.5507511157.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.605.5507511157.kt'>typo_plasma_kotlin--canary.605.5507511157.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.605.5507511157.ts'>typo_plasma_react-native--canary.605.5507511157.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.605.5507511157.swift'>typo_ruler_ios-swift--canary.605.5507511157.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.605.5507511157.kt'>typo_ruler_kotlin--canary.605.5507511157.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.605.5507511157.ts'>typo_ruler_react-native--canary.605.5507511157.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.605.5507511157.swift'>typo_sage_ios-swift--canary.605.5507511157.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.605.5507511157.kt'>typo_sage_kotlin--canary.605.5507511157.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.605.5507511157.ts'>typo_sage_react-native--canary.605.5507511157.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.605.5507511157.swift'>typo_sbermarket_ios-swift--canary.605.5507511157.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.605.5507511157.kt'>typo_sbermarket_kotlin--canary.605.5507511157.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.605.5507511157.ts'>typo_sbermarket_react-native--canary.605.5507511157.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.605.5507511157.swift'>typo_soulmate_ios-swift--canary.605.5507511157.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.605.5507511157.kt'>typo_soulmate_kotlin--canary.605.5507511157.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.605.5507511157.ts'>typo_soulmate_react-native--canary.605.5507511157.ts</a><br>
      </blockquote>
    </details>
  </blockquote>
</details>

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.219.0-canary.605.5507511157.0
  npm install @salutejs/plasma-core@1.124.0-canary.605.5507511157.0
  npm install @salutejs/plasma-hope@1.219.0-canary.605.5507511157.0
  npm install @salutejs/plasma-icons@1.153.0-canary.605.5507511157.0
  npm install @salutejs/plasma-temple@1.168.0-canary.605.5507511157.0
  npm install @salutejs/plasma-ui@1.200.0-canary.605.5507511157.0
  npm install @salutejs/plasma-web@1.219.0-canary.605.5507511157.0
  npm install @salutejs/plasma-sb-utils@0.122.0-canary.605.5507511157.0
  # or 
  yarn add @salutejs/plasma-b2c@1.219.0-canary.605.5507511157.0
  yarn add @salutejs/plasma-core@1.124.0-canary.605.5507511157.0
  yarn add @salutejs/plasma-hope@1.219.0-canary.605.5507511157.0
  yarn add @salutejs/plasma-icons@1.153.0-canary.605.5507511157.0
  yarn add @salutejs/plasma-temple@1.168.0-canary.605.5507511157.0
  yarn add @salutejs/plasma-ui@1.200.0-canary.605.5507511157.0
  yarn add @salutejs/plasma-web@1.219.0-canary.605.5507511157.0
  yarn add @salutejs/plasma-sb-utils@0.122.0-canary.605.5507511157.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-605/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-605/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-605/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-605/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-605/

**Review by Yakutoc — APPROVED** [review]

File: auto-plugins/src/upload-assets-extend.ts [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+            auto.logger.log.success(endent`
+          Download canary assets:
+          
+          ${assets.map((asset: AssetResponse) => `${asset.name}: ${asset.browser_download_url}`).join('  \n')}
+        `);
+
+            const prNumber = getPrNumberFromEnv();
+
+            if (!prNumber || !includeBotPrs) {
```
Comment (Yeti-or) [inline auto-plugins/src/upload-assets-extend.ts:93]: не понял что делает параметр: `includeBotPrs`

**Review by Yeti-or — APPROVED** [review]

File: auto-plugins/src/upload-assets-extend.ts (hunk 1 above)
Comment (neretin-trike) [reply auto-plugins/src/upload-assets-extend.ts:93]: да просто позволяет не добавлять сообщения в PullRequest. Для случаев, когда не надо загружать их канарейки и брать только из релиза

**Pull request merged**